fix: add migration to remove old ckeditor table

earthcomfy opened the pull request:
This PR attempts to add a migration file to remove the old ckeditor table.
Fixes #60 

## Summary by Sourcery

Bug Fixes:
- Removes a stale database table from a previous version of the application to avoid conflicts.

earthcomfy commented:
I used schema_editor because it provides a Django-agnostic way to manage database schema changes. Also, the previous migration file uses it.

DmytroLitvinov commented:
HI there,
Should it be really in the package's migration instead of mentioning it in case someone is migrated? 

My concern is that it automatically drops the tables.

earthcomfy commented:
@DmytroLitvinov I brought this idea in the [issue](https://github.com/django-cms/djangocms-text/issues/60) and thought we wanted to move forward with this approach. @fsbraun

fsbraun commented:
@DmytroLitvinov Any regular migration would replace the old model by the new one retaining already existing data (if possible). This is also the case for this migration. I would second adding it. (Thanks, @earthcomfy )

It might be worthwhile mentioning that one can avoid the old tables to be dropped if migrations are run up to `0003` and `0004` is faked. Maybe we can add this to the migration guide in the readme? The issue of doing this will be potential constraints.

fsbraun commented:
@earthcomfy Why is the `CASCADE` command important for Postgres? Can you comment how you tested this (with Postgres and which databases)?

earthcomfy commented:
@fsbraun I tested this by creating a test project inside my branch. I first added `djangocms_text_ckeditor` into my installed apps and when I run migrate it creates the `djangocms_text_ckeditor_text` table. Then I replaced the installation with `djangocms_text` and run `migrate`. This applies all the migration files including the drop table and the table no longer exists.

I also tested this both with sqlite and postgres.

I added cascade because of the description mentioned in the [issue](https://github.com/django-cms/djangocms-text/issues/60). In PostgreSQL, to drop a table that is referenced by a view or a foreign-key constraint of another table, we have to use cascade. That is my thought process.

fsbraun reviewed (approved):
Hey @earthcomfy ! Thanks for the good work!!

fsbraun commented:
@earthcomfy After @DmytroLitvinov 's comment I am considering to just drop any foreign key constraints to the `djangocms_text_ckeditor_text` table. This would avoid deleting data AND solve the problem. Since it is a bit late for today, I'll see what I can do tomorrow.

DmytroLitvinov commented:
Started to migrate project from `djangocms-text-ckeditor` to `djangocms-text` and just deleted the table and it works also 👍
